pkg/services/annotations/annotations.go

```diff
@@ -7,10 +7,12 @@ import (
 	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/services/user"
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util/errutil"
 )
 
 var (
 	ErrTimerangeMissing = errors.New("missing timerange")
+	ErrBaseBadRequest   = errutil.NewBase(errutil.StatusBadRequest, "annotations.bad-request")
```

Comment:
Make this more specific, it's better to have `ErrBaseTagLimitExceeded`, as you'd expect one unique message layout per base (for internationalization purposes). I guess in this case you probably also want to tell the caller that that was the cause, so add `errutil.WithPublicMessage` to it as well to make sure the user gets a good message.

You also want to switch out the `Error` in `pkg/api` for the caller to `ErrWithFallback` (literal word substitution there, they're compatible with each other)

Maybe something like this?

```suggestion
	ErrBaseTagLimitExceeded  = errutil.NewBase(errutil.StatusBadRequest, "annotations.tag-limit-exceeded", errutil.WithPublicMessage("Tags length exceeds the maximum allowed."))
```

Reply:
> Make this more specific, it's better to have ErrBaseTagLimitExceeded, as you'd expect one unique message layout per base (for internationalization purposes).
GTK, I had the impression that it should be one base per status code.

> I guess in this case you probably also want to tell the caller that that was the cause, so add errutil.WithPublicMessage to it as well to make sure the user gets a good message.
If we provide a public message do we still need [the custom](https://github.com/grafana/grafana/blob/ceed68e4c4f749f14cdd8d5880b08022a9a95689/pkg/services/sqlstore/annotation.go#L393) one?

Reply:
> If we provide a public message do we still need [the custom](https://github.com/grafana/grafana/blob/ceed68e4c4f749f14cdd8d5880b08022a9a95689/pkg/services/sqlstore/annotation.go#L393) one?

Yes, that message is for Go, this message is for end-users (it might seem like it's the same thing, but we essentially want to start separating those because the contents of what'd be a good error message is different depending on if you're facing the end-user or an operator/developer)